### PR TITLE
OP-1188  |  Delete Inventory

### PR DESCRIPTION
**Pull request by loique70**
See OP-1188.

Add deletion and test  function.

### CODE REVIEW

File: src/main/java/org/isf/medicalinventory/service/MedicalInventoryIoOperation.java [hunk 1]
```diff
@@ -198,4 +202,32 @@ public MedicalInventory getInventoryById(Integer inventoryId) throws OHServiceEx
...
 		return null;
 	}
+
+	/**
+	 * Marks an inventory as deleted by changing its status.
+	 * @param inventoryId - the ID of the inventory to delete.
+	 * @throws OHServiceException if an error occurs during the operation.
+	 */
+	public void deleteInventory(int inventoryId) throws OHServiceException {
+		List<OHExceptionMessage> errors = new ArrayList<>();
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalinventory/service/MedicalInventoryIoOperation.java]: Why use an List?   The way I read the code only one of 3 errors can be generated at any time.  Why not just throw each one as it happens?

File: src/main/java/org/isf/utils/exception/model/OHExceptionMessage.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
  *
  * @author akashytsa
  */
-public class OHExceptionMessage implements Serializable{
+public class OHExceptionMessage extends Throwable implements Serializable{
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/utils/exception/model/OHExceptionMessage.java:33]: This should not be needed.

Comment (loique70) [reply src/main/java/org/isf/utils/exception/model/OHExceptionMessage.java:33]: This is comming from develop branch after the mergin.

File: src/main/java/org/isf/medicalinventory/service/MedicalInventoryIoOperation.java (hunk 1 above)
Comment (loique70) [reply src/main/java/org/isf/medicalinventory/service/MedicalInventoryIoOperation.java]: You hav right, I have applyed the changes as you suggested.

File: src/main/java/org/isf/medicalinventory/service/MedicalInventoryIoOperation.java [hunk 3]
```diff
@@ -22,12 +22,16 @@
 package org.isf.medicalinventory.service;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalinventory/service/MedicalInventoryIoOperation.java:25]: where is this import used in this file?

File: src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java [hunk 4]
```diff
@@ -211,4 +211,13 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
...
 		}
 	}
+
+	/**
+	 * Marks an inventory as deleted by changing its status.
+	 * @param medicalInventory - the medicalInventory of the inventory to delete.
+	 * @throws OHServiceException if an error occurs during the operation.
+	 */
+	public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceException {
+		ioOperations.deleteInventory(medicalInventory);
```
Comment (mwithi) [inline src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java:221]: I think we should also delete new lots created within the inventory, wdyt @ArnaudFonzam ?

Comment (ArnaudFonzam) [reply src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java:221]: absolutely @mwithi

**Comment by mwithi** [comment]
Closing waiting a new PR

**Pull request closed without merging**